src/components/map/MapService.js

```diff
@@ -424,6 +424,24 @@ goog.require('ga_urlutils_service');
           }
           return olLayer;
         };
+
+        // Make a GetLegendGraphic request
+        this.getLegend = function(layer) {
+          var defer = $q.defer();
+          var params = layer.getSource().getParams();
+          var html = '<img src="' + gaUrlUtils.append(layer.url,
+              gaUrlUtils.toKeyValue({
+            request: 'GetLegendGraphic',
+            layer: params.LAYERS,
+            style: params.style || 'default',
+            service: 'WMS',
+            version: params.version || '1.3.0',
+            format: 'image/png',
```

Comment:
It could be 'image/png' and 'image/gif' , but I think we doesn't take too much risk to hardoced the format to 'image/png' . For the GetCapabilities, managing the version needs probably  some work on ol and I don't think that worth it to loose time for this.